src/autopas/containers/verletListsCellBased/verletListsCells/neighborLists/VLCAllCellsNeighborList.h

```diff
@@ -79,6 +80,25 @@ class VLCAllCellsNeighborList : public VLCNeighborListInterface<Particle> {
 
     // Helper function to estimate the number of neighbor lists for one base step
     const auto estimateNumLists = [&](size_t baseCellIndex) {
+      // If the cell is near the end of any dimension we can reduce the estimate to the particles in the base cell,
+      // because all other cells' interactions would be from halos.
+      const auto cellIndex3D = utils::ThreeDimensionalMapping::oneToThreeD(baseCellIndex, cellsPerDim);
+      const auto numTouchedBoundaries = [&]() {
+        int acc = 0;
+        for (size_t i = 0; i < 3; ++i) {
+          // false == 0 ; true == 1
+          acc += static_cast<int>(cellIndex3D[i] == (cellsPerDim[i] - 1));
+        }
+        return acc;
+      }();
+      if (numTouchedBoundaries > 0) {
+        if (useNewton3) {
+          return cells[baseCellIndex].size();
```

Comment:
I'm not so sure of this estimate, but please correct me if my logic is wrong:

- Consider c08, where one boundary is touched. E.g. half of the 2x2x2 cube is across the $z$-boundary.
- The diagonal cell interaction along the $x$-$y$ plane that doesn't include the base cell is still a normal owned interaction, so lists will still be built for these interactions, but these are not included in the estimate.

If this is correct, then for the `numTouchedBoundaries==1` case, you would want `cells[baseCellIndex].size() + cells[baseCellIndex + offset].size() * factor[edge]` (if you'll forgive the crappy pseudocode) where `offset` is the offset for the cell which would store the list for the diagonal in the N3 case.

For the  `numTouchedBoundaries==2` case, I think this is still correct.

Reply:
When newton3 is enabled, there will not be lists for every particle in the base cell. Is it a precise heuristic? Nope. But it seems to be "good enough" not to trigger reallocations because the lion's share of lists comes from the base cell as soon as there is at least one touched boundary. Your suggestion is probably safer because it is more conservative but you would also have to figure out which boundary you touch and which offsets are impacted by this and how much. When I saw that there are no reallocations I just couldn't be bothered anymore because I feel these heuristics are really complicated :sweat_smile: 
So my suggestion is leaving it like it is, adding a TODO and if it ever shows up in a profiler we come back to it :innocent:

Reply:
Suggestion sounds good